controllers/flinkcluster_submit_job_script.go

```diff
@@ -142,10 +143,17 @@ function submit_job() {
         return 1
     fi
 
-    # On success, write job ID and log.
+    # write job ID if there is one
     write_term_log "jobID: ${job_id}"
-    write_term_log_msg "Successfully submitted!" "submit_log"
 
+    # check the job's exit code
+    if [ $job_exit_code -ne 0 ]; then
+        write_term_log_msg "Job failed with a non-zero exit code: ${job_exit_code}" "submit_log"
+        return 2
```

Comment:
@yolgun why not return 1 as well? 2 normally means shell misuse.

Reply:
changed it